docs/semantics.md

```diff
@@ -402,8 +412,8 @@ Transient errors can occur in distributed systems like Cloud Storage, such as ne
 ## Missing features
 
 Not all of the usual file system features are supported. Most prominently:
-- Renaming directories is by default not supported. A directory rename cannot be performed atomically in Cloud Storage and would therefore be arbitrarily expensive in terms of Cloud Storage operations, and for large directories would have high probability of failure, leaving the two directories in an inconsistent state.
-- However, if your application can tolerate the risks, you may enable renaming directories in a non-atomic way, by setting ```--rename-dir-limit```. If a directory contains fewer files than this limit and no subdirectory, it can be renamed.
+- Renaming directories is by default not supported in non-hierarchical buckets. A directory rename cannot be performed atomically in these buckets and would therefore be arbitrarily expensive in terms of Cloud Storage operations, and for large directories would have high probability of failure, leaving the two directories in an inconsistent state. This can be resolved using cloud storage buckets with hierarchical namespace enabled. HNS enabled buckets support atomic folder renames and are fully compatible with Cloud Storage FUSE, offering seamless integration.
```

Comment:
Can we use same terminology here regarding non-heirarchical buckets. Its mentioned as Heierarchical namespace buckets above. We can use similar here - "Non Heirarchical Namespace Buckets"  with a link to HNS feature

Reply:
I suggest rewording:


Renaming directories is only supported in Hierarchical Namespace Buckets, where they are fast and atomic. Renaming directories in flat namespace buckets is by default not supposed. A directory rename cannot be performed atomically in these flat buckets and would therefore be arbitrarily expensive in terms of Cloud Storage operations, and for large directories would have high probability of failure, leaving the two directories in an inconsistent state.

Reply:
Done